include/treelite/tree_impl.h

```diff
@@ -348,6 +355,18 @@ inline void InitArrayFromPyBuffer(ContiguousArray<T>* vec, PyBufferFrame buffer)
   vec->UseForeignBuffer(buffer.buf, buffer.nitem);
 }
 
+inline void InitScalarFromPyBuffer(TypeInfo* scalar, PyBufferFrame buffer) {
```

Comment:
Is it possible to pass this by reference to avoid a copy?

```suggestion
inline void InitScalarFromPyBuffer(TypeInfo* scalar, const PyBufferFrame& buffer) {
```

<br>

( I'm guessing there are other lines that would be updated if we change this as well. Though I'm guessing you know that much better than I 😄 )

Reply:
No deep copy is involved when we copy `PyBufferFrame`, since it holds memory references to existing objects. Do you think we should still avoid copying it?

Reply:
Thanks for the info. I don't have deep thoughts here. It was just something that occurred to me when looking at the code. It sounds like you know better here. Feel free to mark as resolved 🙂